File: parking_lot_system/parking_lot/vehicle_factory.py
from .vehicle import Car, Motorcycle, Truck

class VehicleFactory:
    @staticmethod
    def create_vehicle(vehicle_type, license_plate):
        if vehicle_type == "car":
            return Car(license_plate)
        elif vehicle_type == "motorcycle":
            return Motorcycle(license_plate)
        elif vehicle_type == "truck":
            return Truck(license_plate)
        else:
            raise ValueError("Unknown vehicle type")

File: parking_lot_system_go/parkinglot/vehicle_factory.go
package parkinglot

import "fmt"

func VehicleFactory(vehicleType, licensePlate string) Vehicle {
	switch vehicleType {
	case "car":
		return NewCar(licensePlate)
	case "motorcycle":
		return NewMotorcycle(licensePlate)
	case "truck":
		return NewTruck(licensePlate)
	default:
		panic(fmt.Sprintf("Unknown vehicle type: %s", vehicleType))
	}
}
